### PR TITLE
Refactor IsPowerVSZoneSupportsPER() code to use interface

### LINKED ISSUE #1774: Refactor IsPowerVSZoneSupportsPER() to use interface
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Currently [ IsPowerVSZoneSupportsPER()](https://github.com/Karthik-K-N/cluster-api-provider-ibmcloud/blob/81e93e2aa11f6ce94ff4df941720b02224f812d4/cloud/scope/powervs_cluster.go#L589-L614) is directly calling dataceter client to get details, It will be problem while implementing UT so consider refactoring it to use interfaces which will help in mocking.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Shilpa-Gokul**
**What this PR does / why we need it**:
Currently IsPowerVSZoneSupportsPER() uses a get call directly to fetch datacenter information, this might be a problem to mock when we have to add unit tests, hence refactoring the code to use interface.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1774

### CODE REVIEW

File: pkg/cloud/services/powervs/service.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ package powervs
 
 import (
 	"context"
+	"fmt"
+	"github.com/IBM-Cloud/power-go-client/power/client/datacenters"
```
Comment (Karthik-K-N) [inline pkg/cloud/services/powervs/service.go:22]: Lets fix the import order like other files.

**Review by Karthik-K-N — COMMENTED** [review]
Thank you, Overall LGTM, Lets fix the failing test.

File: pkg/cloud/services/powervs/service.go (hunk 1 above)
Comment (Shilpa-Gokul) [reply pkg/cloud/services/powervs/service.go]: Addressed

**Review by Amulyam24 — COMMENTED** [review]
/lgtm

**Review by Karthik-K-N — COMMENTED** [review]
/lgtm

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: pkg/cloud/services/powervs/service.go [hunk 2]
```diff
@@ -185,3 +187,17 @@ func (s *Service) GetNetworkByName(networkName string) (*models.NetworkReference
...
+	// though the function name is WithDatacenterRegion it takes zone as parameter
+	params := datacenters.NewV1DatacentersGetParamsWithContext(context.TODO()).WithDatacenterRegion(zone)
+	datacenter, err := s.session.Power.Datacenters.V1DatacentersGet(params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get datacenter details for zone: %s err:%w", zone, err)
+	}
+	if datacenter == nil || datacenter.Payload == nil || datacenter.Payload.Capabilities == nil {
+		return nil, fmt.Errorf("failed to get datacenter capabilities for zone: %s", zone)
+	}
+	return datacenter.Payload, nil
```
[the remark this answers is not in the document]
Comment (Shilpa-Gokul) [reply pkg/cloud/services/powervs/service.go]: Done

File: pkg/cloud/services/powervs/service.go [hunk 3]
```diff
@@ -185,3 +187,17 @@ func (s *Service) GetNetworkByName(networkName string) (*models.NetworkReference
 
 	return network, nil
 }
+
+// GetDatacenters fetches the datacenter capabilities for the given zone.
+func (s *Service) GetDatacenters(zone string) (*models.Datacenter, error) {
```
[the remark this answers is not in the document]
Comment (Karthik-K-N) [reply pkg/cloud/services/powervs/service.go]: It should be zone, Found this conversation which mention to use zone https://github.com/openshift/installer/pull/7736#discussion_r1408120957

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**